### DISCUSSION TITLE
Dynamic Lighting Support (Windows 11)

### DISCUSSION THREAD

**Discussion by Tanmoy2504** [BartoszCichecki/LenovoLegionToolkit#992] [Ideas]
### Rules

- [X] I made myself familiar with the README.
- [X] I checked closed issues for similar ones and did not find a similar feature request mentioned before.
- [X] I understand that, if insufficient information was provided, my issue will be closed without an answer.

### Version

2.16.2

### OS

Windows 11 Build 22621.2361

### Device

Legion 5 Pro 16ARH7H (82RG)

### Is your feature request related to a problem?

No, the requested feature is not associated with any problem whatsoever

### How would you like the problem to be solved?

Implementation of this feature (if possible, not mandatory)

### What alternatives have you considered?

None

### Additional information

Windows 11 Build 22621.2361 has introduced "Dynamic Lighting", their own version of RGB equipment control (Settings > Personalization > Dynamic Lighting). Although the current RGB Controlled from LegionToolkit is good, I wanted to know if support for this is possible. Dynamic Lighting utilises "HID LampArray"  and as per the developer article from microsoft which i've linked below, can be used to customize other peripherals even arduino based custom peripheral control.  [Dynamic Lighting Guide](https://learn.microsoft.com/en-us/windows-hardware/design/component-guidelines/dynamic-lighting-devices)

while this is a feature suggestion, I want the Toolkit working in tandem with this. Since it requires either native FW support (from oem) or a custom driver utilizing Virtual HID Framework (VHF). All the details and examples are provided in the above link. 

As I mentioned earlier, most of the things Dynamic Lighting can currently do can & is done by Legion Toolkit but it offers features such as syncing with desktop themes, more effects (although some won't work with 4 zone and is intended for per key), brightness slider and ability for apps (ex. spotify or an xbox game) to match the content and rgb colors in future. Since LLT uses its own drivers and control, I wanted you to consider this in future

<img width="773" alt="Screenshot 2023-10-04 215904" src="https://github.com/BartoszCichecki/LenovoLegionToolkit/assets/37358272/68d50a9a-e8e6-4bd4-87db-bb674c0c8c73">

**Comment by BartoszCichecki** [comment]
Converting this issue to a Discussion. This idea is outside of scope of this project, and while fairly easy to implement it requires creating of a driver which needs to be signed to be usable by end users, which is quite expensive.

**Comment by Lampadina17** [comment]
Just read windows values from LTT and set kb color from LTT, i don't think is needed a kernel driver only for reading a value

**Reply by BartoszCichecki** [reply]
"I don't think" is neither useful
nor enough. You either know how to do it and provide proof of concept or you don't.
